runtime/common/marshaling/marshal_errors.ts

```diff
@@ -0,0 +1,103 @@
+import {MarshalingInjectedKeys} from './constants';
+import {StatusCodeError} from '../../../api';
+
+enum ErrorClassType {
+  System = 'System',
+  Coda = 'Coda',
+  Other = 'Other',
+}
+
+const recognizableSystemErrorClasses: ErrorConstructor[] = [
+  Error,
+  EvalError,
+  RangeError,
+  ReferenceError,
+  SyntaxError,
+  TypeError,
+  URIError,
+];
+
+const recognizableCodaErrorClasses: ErrorConstructor[] = [
+  // StatusCodeError doesn't have the new StatusCodeError(message) constructor but it's okay.
+  StatusCodeError as any,
+];
+
+function getErrorClassType(err: Error): ErrorClassType {
+  if (recognizableSystemErrorClasses.some(cls => cls === err.constructor)) {
+    return ErrorClassType.System;
+  }
+
+  if (recognizableCodaErrorClasses.some(cls => cls === err.constructor)) {
+    return ErrorClassType.Coda;
```

Comment:
Any particular reason the Coda errors need to be handled differently than the system errors?

Reply:
oh, theoretically, we can create a Coda error class called URIErrorr. 

actually I first separated them because there's one system error called InternalError. but then I realized that it was non-standard. 

let me simplify this a bit.

Reply:
actually. let's leave this as is. I feel like it's more clear to have a separate namespace for coda errors.